src/execution/execute-afm.ts

```diff
@@ -202,14 +202,20 @@ export function mergePage(
     const result = prevExecutionResult;
     const { headerItems, data, paging } = executionResult;
 
+    function mergeHeaderItemsForEachAttribute(dimension: number) {
```

Comment:
please move this outside mergePage function and avoid of using function side-effects (modifying result object)